osbs/build/spec.py

```diff
@@ -354,6 +357,23 @@ def set_params(self, git_uri=None, git_ref=None,
         self.yum_proxy.value = yum_proxy
         self.koji_parent_build.value = koji_parent_build
 
+        odcs_enabled = True if odcs_url and odcs_openidc_secret else False
+        if (signing_intent or compose_ids) and not odcs_enabled:
+            raise OsbsValidationException(
+                'signing_intent and compose_ids are allowed only when ODCS is enabled')
+
+        if signing_intent and compose_ids:
+            raise OsbsValidationException(
+                'Please only define signing_intent -OR- compose_ids, not both')
+
+        if compose_ids and yum_repourls:
+            raise OsbsValidationException(
+                'Please only define yum_repourls -OR- compose_ids, not both')
+
+        if not (compose_ids is None or isinstance(compose_ids, list)):
```

Comment:
Is this needed? In theory, it could be a tuple, a generator, etc.

Reply:
I was following that yum_repourls pattern

Reply:
Hmm I'll let others chime. Don't feel strongly about it either way.

Reply:
ok

Reply:
The `yum_repourls` case is only complicated because strings are also iterable. How about checking that `iter(compose_ids)` doesn't raise `TypeError`?

Reply:
right I can do that